### PR TITLE
Enable some small-batch optimization in tests

### PR DESCRIPTION
**Pull request by bkirwi**
### Motivation

@guswynn pointed out that this is a fruitful way to test #18307!

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Review by def- — CHANGES_REQUESTED** [review]
Any negative side effect of setting this for all mzcompose tests?

File: misc/python/materialize/mzcompose/services.py [hunk 1]
```diff
@@ -118,6 +118,7 @@ def __init__(
             f"--default-storage-host-size={self.default_storage_size}",
             # We wish to test the new implementation as its the canonical one, but
             # want to control the rollout to production.
```
[the remark this answers is not in the document]
Comment (bkirwi) [reply misc/python/materialize/mzcompose/services.py:121]: Point - updated!

**Review by def- — APPROVED** [review]

**Comment by bkirwi** [comment]
> Any negative side effect of setting this for all mzcompose tests?

Shouldn't be! That is done intentionally - we plan to make this the default everywhere soon, so it's appropriate to run all our tests with it.

**Comment by bkirwi** [comment]
Nightly run: https://buildkite.com/materialize/nightlies/builds/2048

One failure, matching some recent failures in `main`, so I'm content.

**Pull request merged**